From 5e2f27e166e7a8735b80abce51a5c9ca7cb7d26b Mon Sep 17 00:00:00 2001
From: nick <nick@egan.gg>
Date: Tue, 13 Dec 2022 15:51:52 -0500
Subject: [PATCH] use in planner

---
 selfdrive/controls/lib/longitudinal_planner.py | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/selfdrive/controls/lib/longitudinal_planner.py b/selfdrive/controls/lib/longitudinal_planner.py
index a0f631832..7e44b744d 100755
--- a/selfdrive/controls/lib/longitudinal_planner.py
+++ b/selfdrive/controls/lib/longitudinal_planner.py
@@ -3,6 +3,14 @@ import math
 import numpy as np
 from common.numpy_fast import clip, interp
 
+hook_available = False
+try:
+  from opsetspeed.shminject import Hook
+  hook_available = True
+except:
+  pass
+
+
 import cereal.messaging as messaging
 from common.conversions import Conversions as CV
 from common.filter_simple import FirstOrderFilter
@@ -46,6 +54,7 @@ def limit_accel_in_turns(v_ego, angle_steers, a_target, CP):
 
 class LongitudinalPlanner:
   def __init__(self, CP, init_v=0.0, init_a=0.0):
+    self.hook = Hook() if hook_available else None
     self.CP = CP
     self.mpc = LongitudinalMpc()
     self.fcw = False
@@ -83,6 +92,11 @@ class LongitudinalPlanner:
     v_cruise_kph = min(v_cruise_kph, V_CRUISE_MAX)
     v_cruise = v_cruise_kph * CV.KPH_TO_MS
 
+    v_override = self.hook.update(v_cruise) if hook_available else v_cruise
+    if v_override < v_cruise:
+      self.mpc.mode = 'acc'
+      v_cruise = v_override
+
     long_control_off = sm['controlsState'].longControlState == LongCtrlState.off
     force_slow_decel = sm['controlsState'].forceDecel
 
-- 
2.25.1

